### PR TITLE
Update MdBook

### PR DESCRIPTION
**Pull request by tina1998612**
Current content: 
- intro
- how to connect to testnet
- how to become validator 
- how to contribute to the book
- beacon chain intro

Todo: 
- simplify the readme after the book is published

### CODE REVIEW

**Comment by arnetheduck** [comment]
there's quite a few files in here that don't belong - ie `nbc.log` and the submodule updates - let's rebase this branch without these files in the history before merging

File: docs/src/beacon-chain.md [hunk 1]
```diff
@@ -1,4 +1,19 @@
...
+
+## Differences Compared to Ethereum 1.0
+
+In traditional PoW, those that propose new blocks are called **_miners_**, whereas in PoS, they are called **_validators_**. In essence, _miners_ rely on actual hardware (such as some specifically manufactured mining machines), while _validators_ rely on just software.
+
+## What it is Like to Be a Validator?
+
+It is obvious that you must have enough computing power or a dedicated hardware in order to be a miner, but how about being a validator? Here is a breif overview:
+
+1. A special smart contract named **_registration contract_** is deployed on the original Ethereum blockchain. Note that in this case, the new beacon chain and the original blockchain co-exists.
```
Comment (arnetheduck) [inline docs/src/beacon-chain.md:15]:
```suggestion
1. A special smart contract named **_deposit contract_** is deployed on the original Ethereum blockchain. Note that in this case, the new beacon chain and the original blockchain co-exists.
```

File: docs/src/beacon-chain.md [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+2. To "register" as a validator, you have to first deposit **_32 Etheres_** from your account to this smart contract.
```
Comment (arnetheduck) [inline docs/src/beacon-chain.md:16]:
```suggestion
2. To "register" as a validator, you have to first deposit **_32 Ether_** from your account to this smart contract.
```

File: docs/src/beacon-chain.md [hunk 3]
```diff
@@ -1,4 +1,19 @@
...
+
+In short, the beacon chain is a **new type of blockchain** to help the Ethereum blockchain to smoothly transfer its consensus algorithm from PoW (Proof of Work) to PoS (Proof of Stake), aka Ethereum 2.0. You can also see it as a hybrid PoS + PoW solution.
+
+## Differences Compared to Ethereum 1.0
+
+In traditional PoW, those that propose new blocks are called **_miners_**, whereas in PoS, they are called **_validators_**. In essence, _miners_ rely on actual hardware (such as some specifically manufactured mining machines), while _validators_ rely on just software.
+
+## What it is Like to Be a Validator?
+
+It is obvious that you must have enough computing power or a dedicated hardware in order to be a miner, but how about being a validator? Here is a breif overview:
```
Comment (kdeme) [inline docs/src/beacon-chain.md:13]:
```suggestion
It is obvious that you must have enough computing power or dedicated hardware in order to be a miner, but how about being a validator? Here is a brief overview:
```

File: docs/src/beacon-chain.md [hunk 4]
```diff
@@ -1,4 +1,19 @@
...
+
+1. A special smart contract named **_registration contract_** is deployed on the original Ethereum blockchain. Note that in this case, the new beacon chain and the original blockchain co-exists.
+2. To "register" as a validator, you have to first deposit **_32 Etheres_** from your account to this smart contract.
+3. Run the beacon node and wait for the network to sync before your validator is activated.
+4. That's all! Remember to stay connected to the network, or you may lost your deposit. :P
```
Comment (kdeme) [inline docs/src/beacon-chain.md:18]:
```suggestion
4. That's all! Remember to stay connected to the network, or you may lose your deposit. :P
```

File: docs/src/contribute.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Contribute
+
+Follow these steps to contribute to this book! <br>
+We use a utility tool called mdBook to create online books from Markdown files.
```
Comment (kdeme) [inline docs/src/contribute.md:4]:
```suggestion
We use an utility tool called mdBook to create online books from Markdown files.
```

File: docs/src/contribute.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Contribute
+
+Follow these steps to contribute to this book! <br>
```
Comment (kdeme) [inline docs/src/contribute.md:3]: You can get rid of the <br> with double space I blelieve.

File: docs/src/validator.md [hunk 7]
```diff
@@ -1,19 +1,60 @@
...
+git pull
+make update
 make testnet0        # This will build Nimbus and all other dependencies
                      # and connect you to testnet0
 ```
 
-The testnets are restarted once per week, usually on Monday evenings (UTC)) and integrate the changes for the past week.
+<img src="./img/connect_testnet.PNG" alt="" style="margin: 0 40 0 40"/>
+
+7. The testnet should now be up and running. Then, you will be prompted to enter your private key of the account you want to deposit the 32 Etheres from. Find your private key from MetaMask as below:
```
Comment (kdeme) [inline docs/src/validator.md:38]:
```suggestion
7. The testnet should now be up and running. Then, you will be prompted to enter your private key of the account you want to deposit the 32 Ether from. Find your private key from MetaMask as below:
```

File: docs/src/validator.md [hunk 8]
```diff
@@ -1,19 +1,60 @@
...
+
+<img src="./img/export_pkey.PNG" alt="" width="200" style="margin: 0 40 0 40"/>
+
+<img src="./img/enter_private_key.PNG" alt="" style="margin: 0 40 0 40"/>
+
+8. Wait for a few seconds until you see your deposit has been sent:
```
Comment (kdeme) [inline docs/src/validator.md:44]:
```suggestion
8. Wait for a few seconds until you see that your deposit has been sent:
```

File: docs/src/validator.md [hunk 9]
```diff
@@ -1,19 +1,60 @@
...
+
+9. Now you should be syncing with the network. It may take a while (may be quite a few hours). You can know that you are synced if you see the following output.
+
+<img src="./img/success.PNG" alt="" style="margin: 0 40 0 40"/>
+
+You can also get a brief estimate of the time remaining until your network gets synced by comparing the output `epoch` value and the one in the blockchain explorer (the [witti explorer](https://witti.beaconcha.in) for example).
+
+## Trouble Shooting
+
+1. The directory that stores the blockchain data of the testnet is `build/data/testnet0` (replace `testnet0` with other testnet names). Delete this folder if you want to start over. For example, you can start over with a fresh storgae if you entered a wrong private key.
```
Comment (kdeme) [inline docs/src/validator.md:56]:
```suggestion
1. The directory that stores the blockchain data of the testnet is `build/data/testnet0` (replace `testnet0` with other testnet names). Delete this folder if you want to start over. For example, you can start over with a fresh storage if you entered a wrong private key.
```

But there is actual a `make clean-<testnetname> command for each testnet to do this.
Perhaps you want to add something about backing up the private key as this gets removed with the command?

**Comment by kdeme** [comment]
I noticed some accidental nbc.log commit and vendor bumps?

**Comment by tina1998612** [comment]
have pushed on to devel with all the above comments fixed:)

**Comment by tina1998612** [comment]
Online at: https://status-im.github.io/nim-beacon-chain/validator.html

**Pull request closed without merging**